lib/win32/certstore.rb

```diff
@@ -62,7 +63,7 @@ def open(store_name)
       certstore_handler = CertOpenSystemStoreW(nil, wstring(store_name))
       unless certstore_handler
         last_error = FFI::LastError.error
-        raise Chef::Exceptions::Win32APIError, "Unable to open the Certificate Store `#{store_name}` with error: #{last_error}."
+        raise SystemCallError.new("Unable to open the Certificate Store `#{store_name}`.", last_error)

```

Comment:
I want to keep it simple since these are system error https://msdn.microsoft.com/en-us/library/windows/desktop/ms681384(v=vs.85).aspx and want to keep this align with other win32 libraries https://github.com/djberg96/win32-service/blob/ffi/lib/win32/daemon.rb.

Reply:
Okay so keeping the scope of this task in mind, may be we can do it in a separate task if required. We can also think about moving `lookup_error` function accordingly.

@btm , please decide if these changes are required.

Reply:
I agree, it might be good to have a separate class for handling errors.